### DISCUSSION TITLE
Auto break when rendering Table doesnt seem to respect t_margin

### DISCUSSION THREAD

**Discussion by VonArmin** [py-pdf/fpdf2#817] [Q&A] [answered]
Hey,

I might have found an issue when generating a large table that spans over multiple pages.

While the generating works fine and it breaks the page correctly on the bottom margin, it doesnt soom to start at the top margin.

Ive tried setting it to multiple factors of margin values but none seem to change anything. Any ideas?

**Comment by Lucas-C** [comment]
Hi @VonArmin!

Thank you for getting in touch.
Could you please report this as a bug here, with some minimal code that reproduces your problem?
https://github.com/PyFPDF/fpdf2/issues/new/choose

I'd be happy to have a look at it, but I need some starting point as code to reproduce the issue your are facing

**Comment by VonArmin** [comment, marked as answer]
Okay so Ive found a workaround. 

The problem was when using both set_xy in the header method it overrrides the coordinates set in set_top_margin.

I just added an extra set_xy at the end of my header method (since this function will also be called when add_page is called though the auto break process) (note: the header method is called first, followed by setting the self.x and self.y to the left and top margins, why that isnt working as expected, im not sure)

I can still provide a reproduction later if necessary.

**Reply by Lucas-C** [reply]
Perfect if you found a solution to your issue!

Thank you for sharing it there 😊
